tests/test_es_etymology.py

```diff
@@ -0,0 +1,98 @@
+                        {
+                            "args": {
+                                "leng": "de",
+                            },
+                            "name": "etimología",
+                        },
+                    ],
+                },
+            },
+            {
+                "input": "{{etimología|catalán|bagre}}, y este del latín ''[[pargus]]'', a su vez del griego ''[[φάγρος]]'' (''phágros'')",
+                "expected": {
+                    "etymology_templates": [
+                        {
+                            "args": {
+                                "1": "catalán",
+                                "2": "bagre",
+                            },
+                            "expansion": "Del catalán bagre",
+                            "name": "etimología",
+                        },
+                    ],
+                    "etymology_text": "Del catalán bagre, y este del latín pargus, a su vez del griego φάγρος (phágros)",
+                },
+            },
+            {
+                "input": "{{etimología2|De ''[[hibridar]]'', y de ''[[-ción]]''}}",
+                "expected": {
+                    "etymology_templates": [
+                        {
+                            "args": {
+                                "1": "De hibridar, y de -ción",
+                            },
+                            "expansion": "De hibridar, y de -ción",
+                            "name": "etimología2",
+                        },
+                    ],
+                    "etymology_text": "De hibridar, y de -ción",
+                },
+            },
+        ]
+        for case in test_cases:
+            with self.subTest(case=case):
+                self.wxr.wtp.start_page("")
+                page_data = self.get_default_page_data()
+
+                root = self.wxr.wtp.parse(case["input"])
+
+                process_etymology_block(
+                    self.wxr, page_data[0], root
+                )
+
+                result = page_data[0].model_dump(exclude_defaults=True)
+
+                # XXX The commented-out lines currently fail because the WiktextractContext needs to know the templates.
+
+                # self.assertEquals(case["expected"].get("etymology_text"), result.get("etymology_text"))
+
+                result_etymology_templates = result.get("etymology_templates")
+                expected_etymology_templates = case["expected"].get("etymology_templates")
+
+                self.assertTrue(len(result_etymology_templates) == len(expected_etymology_templates))
+                for i, expected_template in enumerate(expected_etymology_templates):
+                    result_template = result_etymology_templates[i]
+
+                    self.assertEquals(expected_template["name"], result_template["name"])
+                    if "args" in result_template:
+                        self.assertDictEqual(expected_template["args"], result_template["args"])
```

Comment:
```suggestion
                    self.assertEqual(expected_template["name"], result_template["name"])
                    if "args" in result_template:
                        self.assertEqual(expected_template["args"], result_template["args"])
```

`assertEquals` is removed in Python 3.12 and `assertEqual` calls `assertDictEqual` when passed dictionaries.

You could use [`Wtp.add_page()`](https://github.com/tatuylonen/wikitextprocessor/blob/619981b7f478d2d1479494ca69e1796865afe134/src/wikitextprocessor/core.py#L778-L786) to add template with expanded body in test:


```python
self.wxr.wtp.add_page("Template:name", 10, "expanded text")
```

Reply:
How about now? I couldn't get the template to actually expand its parameters so I hardcoded a generic expansion for the templates, like other tests do. Also, I found and fixed a flaw in the logic thanks to the test 😅